code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm

```diff
@@ -767,7 +777,7 @@ var/list/cult_spires = list()
 	..()
 
 /obj/structure/cult/spire/proc/upgrade()
-	var/new_stage = Clamp(1, veil_thickness, 3)
+	var/new_stage = Clamp(veil_thickness, 1, 3)
```

Comment:
I feel like the math is wrong, or at least, hard to read.

If I translate ```dm
#define Clamp(x, y, z) 	(x <= y ? y : (x >= z ? z : x))
```
Into a system, I have
```
f(y) = x if (x <= y)
f(y) = y if (x > y and x <= z)
f(y) = z if (x > z)
```

If `f(x)` is `new_stage` and `x` is `veil_thickness`
Currently
```
f(x) = 1 if (x < 1)
f(x) = x if (x >= 1 and x < 3)
f(x) = 3 if (x >= 3)
```
With the change
```
f(x) = x if (x <= x) [Trivially true]
f(x) = 1 if (x <= x and x < 3) [Frist part is trivially true]
f(x) = 3 if (x >= 3)
```
So this will never reach 2 or something ?

Reply:
Ignore me, it's the other way around, previous writing wasn't working and the one in the PR does.

Reply:
Yep. It caused shit like veil thickness of 5 to turn spires invisible.